### PR TITLE
bridge: Add "owned" message to dbus-json3

### PR DESCRIPTION
**Pull request by petervo**
Send a 'owned' message up the channel whenever the dbus name owner changes.

### CODE REVIEW

**Comment by mvollmer** [comment]
Test failures are known and unrelated.

**Comment by stefwalter** [comment]
Sorry for not reviewing this earlier. But I have several questions about this:

 * Why aren't we transferring the unique name in the message?
 * Do we need ```track: true``` once this goes in?

**Comment by petervo** [comment]
1) It seemed to me that was unnecessary information, why would any client need to know who the owner is. Seems to me the important part is if it is owned or not.
2) It's different from track, though clients could implement the track behavior with this. Removing track would break backwards compatibility

**Comment by stefwalter** [comment]
> 1) It seemed to me that was unnecessary information, why would any client need to know who the owner is. Seems to me the important part is if it is owned or not.

I think that callers might want to track a owner that regains ownership of a name.

> 2) It's different from track, though clients could implement the track behavior with this. Removing track would break backwards compatibility

We don't yet guarantee backwards compatibility for all aspects of the protocol. Only those that break adding servers from one version to another.

**Comment by petervo** [comment]
Does that really happen in practice? Isn't it almost always the case that once the name is the released the connection is closed and reacquiring the name even by the same process would be new connection with a new name?

At any rate I can add this, are you still ok with the message being named 'owned'?

I still think the intent here is different enough from track for both to be useful. But if you disagree and want to remove track that can be a separate pull request.

**Comment by stefwalter** [comment]
> Does that really happen in practice? Isn't it almost always the case that once the name is the released the connection is closed and reacquiring the name even by the same process would be new connection with a new name?

It happens rarely in practice. But when it happens, it's usually on the same connection. DBus connections are generally not closed until the process actually exits.

> At any rate I can add this, are you still ok with the message being named 'owned'?

I would suggest ```"owner"```

> I still think the intent here is different enough from track for both to be useful. But if you disagree and want to remove track that can be a separate pull request.

The track stuff is a work around for suboptimal services. If "owner" can do the same thing, then I would suggest implementing it that way in javascript, rather than implementing it in the bridge. We would just set all proxies to ```.valid = false``` when ```track = true``` and ```owner null```.

File: pkg/base1/cockpit.js [hunk 1]
```diff
@@ -1596,8 +1596,8 @@ function full_scope(cockpit, $, po) {
             } else if (msg.meta) {
                 ensure_cache();
                 $.extend(cache.meta, msg.meta);
-            } else if (msg.owned !== "undefined") {
-                $(self).triggerHandler("owned", [ msg.owned ]);
+            } else if (msg.owner !== "undefined") {
```
Comment (stefwalter) [inline pkg/base1/cockpit.js]: Why does this work? It seems like this event will be emitted for each message received.

File: src/bridge/cockpitdbusjson.c [hunk 2]
```diff
@@ -1844,7 +1842,7 @@ on_name_vanished (GDBusConnection *connection,
   CockpitDBusJson *self = COCKPIT_DBUS_JSON (user_data);
   CockpitChannel *channel = COCKPIT_CHANNEL (self);
 
-  send_owned (self, FALSE);
+  send_owned (self, "");
```
Comment (stefwalter) [inline src/bridge/cockpitdbusjson.c]: Lets use null here. It's a valid JSON type.

**Comment by stefwalter** [comment]
This also needs documentation in ```doc/protocol.md````

And what are we going to do with ```track = true```. Should we move it to client side implementation?

**Comment by petervo** [comment]
track is a separate PR, coming soon

**Comment by petervo** [comment]
Fixed and added paragraph to protocol.md

**Pull request closed without merging**